micrometer-core/src/main/java/io/micrometer/core/instrument/config/NamingConvention.java

```diff
@@ -81,10 +81,10 @@ private String toCamelCase(String value) {
                     conventionName.append(str);
                 } else {
                     final char firstChar = str.charAt(0);
-                    if (Character.isTitleCase(firstChar)) {
+                    if (Character.isUpperCase(firstChar)) {
```

Comment:
What was the motivation to switch to using the `*UpperCase` methods instead? Reading the JavaDoc, it seems there is a difference between the behavior of these two methods with some characters (e.g. `Ǉ`, `ǈ`). Using title-case seems more correct to me for camelCase. Whether we need to support the limited characters that would be problematic or not, I'm not sure, though.
https://www.compart.com/en/unicode/U+01C8

Reply:
Looking into it more, I see a single uppercase letter character returns `false` for `isTitleCase`. So currently, for almost all cases, it is going to the else block here.

Reply:
@shakuzen I just wanted to align it with [the `upperCamelCase` implementation](https://github.com/micrometer-metrics/micrometer/blob/1.0.x/micrometer-core/src/main/java/io/micrometer/core/instrument/config/NamingConvention.java#L95-L115) and it seemed reasonable to me to use them as I thought meter names and tag keys should be alphanumeric except delimiters. If my assumption is incorrect, this change should be reverted.

Reply:
`isTitleCase` is definitely not what we want, now that I understand what it does.